membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -41,8 +44,10 @@ class AccountController extends LazyLogging {
       user <- EitherT(Future.successful( maybeUserId \/> "no identity cookie for user"))
       sfUser <- EitherT(tp.contactRepo.get(user).map(_.flatMap(_ \/> s"no SF user $user")))
       subscription <- EitherT(tp.subService.current[P](sfUser).map(_.headOption).map (_ \/> s"no current subscriptions for the sfUser $sfUser"))
+      account <- EitherT(tp.zuoraService.getAccount(subscription.accountId).map(x => Some(x)).map(_ \/> s"no account for subscription: ${subscription.name} with account id ${subscription.accountId}"))
```

Comment:
need a recover here not a map (the Some will never be turned into a None so the \/> will never be an error)

Reply:
the second map needs to be a recover with a left, and the first one needs to map to a right